subprojects/plugins/src/integTest/groovy/org/gradle/java/fixtures/KotlinTestFixturesIntegrationTest.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.java.fixtures
+
+import spock.lang.Issue
+
+/**
+ * Integration tests for the `java-test-fixtures` plugin that involve `kotlin` projects.
+ */
+class KotlinTestFixturesIntegrationTest extends AbstractTestFixturesIntegrationTest {
+    /**
+     * Ensure we make an exception for test fixtures when checking for redundant
+     * configuration usage activation - this combination of plugins should not warn.
+     */
+    @Issue("https://github.com/gradle/gradle/pull/24271")
+    def "test kotlin + java-test-fixtures"() {
+        given:
+        settingsFile << """
+            include 'sub'
+        """
+
+        file("sub/build.gradle") << """
+            plugins {
+                id 'java-library'
+                id 'java-test-fixtures'
+            }
+        """
+
+        addPersonDomainClass("sub", "java")
+        addPersonTestFixture("sub", "java")
+
+        buildFile.text = """
+            plugins {
+                id("org.gradle.java-test-fixtures")
+                id("org.jetbrains.kotlin.jvm") version "1.8.20"
```

Comment:
I was under the impression that we should be keeping all kotlin testing in the smoke test bucket, since we are testing integrations with a third-party plugin here. It seems odd to have this in an integ test. 

Though after some code searching it seems we do not do this in practice. 

In any case, I'm not sure I love the idea of hard-coding the kotlin version here